src/DocFinder.php

```diff
@@ -5,17 +5,24 @@
 namespace Codelicia\Xulieta;
 
 use Symfony\Component\Finder\Finder;
+use function array_map;
 use function basename;
 use function dirname;
 use function is_dir;
+use function sprintf;
 
 final class DocFinder
 {
     private string $directoryOrFile;
 
-    public function __construct(string $directoryOrFile)
+    /** @var string[] */
```

Comment:
would be more specific here: `@var array<int, string>`, or even better `@psalm-var list<string>`